app/code/Bulbulatory/Recomendations/Model/Total/Quote/Custom.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Bulbulatory\Recomendations\Model\Total\Quote;
+
+use Magento\Quote\Model\Quote\Address\Total\AbstractTotal;
+use Magento\Framework\Pricing\PriceCurrencyInterface;
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Api\Data\ShippingAssignmentInterface;
+use Magento\Quote\Model\Quote\Address\Total;
+use Bulbulatory\Recomendations\Block\Recommendation\Index;
+
+class Custom extends AbstractTotal
+{
+   protected $_priceCurrency;
+   protected $index;
+
+   public function __construct(
+      PriceCurrencyInterface $priceCurrency,
+      Index $index
+   ){
+       $this->_priceCurrency = $priceCurrency;
+       $this->index = $index;
+   }
+
+   public function collect(
+       Quote $quote,
+       ShippingAssignmentInterface $shippingAssignment,
+       Total $total
+   )
+   {
+        parent::collect($quote, $shippingAssignment, $total);
+
+        $percent = $this->index->calculateDiscountPercent();
+        $baseDiscount = $total->getSubtotal() * $percent / 100;
+
+        $discount =  $this->_priceCurrency->convert($baseDiscount);
+        $total->addTotalAmount('customdiscount', -$discount);
+        $total->addBaseTotalAmount('customdiscount', -$baseDiscount);
+        $total->setBaseGrandTotal($total->getBaseGrandTotal() - $baseDiscount);
+        $quote->setCustomDiscount(-$discount);
+        return $this;
+    }
+
+    public function fetch(Quote $quote, Total $total) 
+    {
+        $percent = $this->index->calculateDiscountPercent();
+        $discount = $total->getSubtotal() * $percent / 100;
+        return [
+            'code' => $this->getCode(),
+            'title' => $this->getLabel(),
+            'value' => -$discount  
+        ];
+  	  }
```

Comment:
Tutaj też warto sprawdzić, czy w ogóle jakaś zniżka się należy i jeśli nie to zwrócić null, ponieważ obecnie jeśli jej nie ma, to w podsumowaniu w koszyku wyświetla się informacja o zniżce wynoszącej 0.